enterprise_access/utils.py

```diff
@@ -245,3 +245,27 @@ def _days_from_now(days_from_now=0, date_format=None):
     if not date_format:
         return date
     return date.strftime(date_format)
+
+
+def get_course_run_metadata_for_assignment(assignment, content_metadata):
+    """
+    Retrieves metadata for a specific course run associated with an assignment. If the assignment has
+    a preferred course run, returns the metadata for that run. If the preferred run metadata is not 
+    found, returns normalized_metadata.
+
+    Args:
+        assignment (dict): The assignment object.
+        content_metadata (dict): The content metadata object.
+
+    Returns:
+        dict: Course run metadata if available, otherwise normalized_metadata.
+    """
+    if preferred_course_run_key := assignment.preferred_course_run_key:
+        course_runs = content_metadata.get('course_runs', [])
+        for course_run in course_runs:
+            if course_run.get('key') == preferred_course_run_key:
+                # manually applying existing key value mappings for consistent return structure
+                course_run['start_date'] = course_run.get('start')
+                course_run['end_date'] = course_run.get('end')
+                return course_run
+    return content_metadata.get('normalized_metadata', {})
```

Comment:
Great job including a fallback, but I would return the advertised course run's course run metadata as opposed to just normalized metadata.

Could be something along the lines of:
```py
def get_course_run_metadata_for_assignment(assignment, content_metadata):
  if preferred_course_run_key := assignment.preferred_course_run_key:
      ....
  advertised_course_run_uuid = content_metadata.get('advertised_course_run_uuid`);
  return next(course_run for course_run in content_metadata["course_runs"] if course["uuid"] == advertised_course_run_uuid)
  ```

Reply:
Perfect! Adam also shared the same thought above and I wasn't sure what should be default return. Thanks for sharing!... just pushed it.